src/reconstruction/ObjectModel.jl

```diff
@@ -234,3 +241,180 @@ function add_biomass_metabolite(
 
     m
 end
+
+"""
+$(TYPEDSIGNATURES)
+
+To `biomass_rxn_id` in `model`, add a pseudo-isozyme that approximates the
+effect ribosome synthesis has on growth.
+
+# Bacterial growth law models
+Numerous experimental studies have shown that during steady state growth the
+cellular density of E. coli (and probably other bacteria) is constant. As a
+consequence of this, growth law models typically assume that the total proteome
+capacity (mass fraction of protein in the cell) is limited. Further, it has been
+shown experimentally that the ribosomal protein content of a bacterial cell
+increases with faster growth rate. Ribosomes are used to make proteins (and
+themselves), leading to a trade-off: faster growth requires more ribosomes to
+make more enzymes to grow, but this reduces the amount of proteome space "left
+over" for biosynthetic enzymes. See Mori, Matteo, et al. "Constrained allocation
+flux balance analysis." PLoS computational biology 12.6 (2016) for more details.
+
+# Implementation
+By adding a protein cost to biomass synthesis this effect can be simulated. The
+parameter `weight` needs to be estimated from data, but acts like a turnover
+number. Lower `weight` means more ribosome is required for growth (`ribosome =
+growth/weight`). The molar mass of the ribosome is `1`.
+
+# Note
+1. This modifications makes the underlying biomass reaction unidirectional.
+2. The `pseudoribosome_id` defaults to `pseudoribosome` and must be manually included
+   in any capacity bound later used in enzyme constrained models.
+3. This modification also adds a pseudogene called `pseudoribosome_id`.
+
+The pseudo-isozyme acts like a regular gene product,
+```
+ribosome = weight * biomass_flux
+```
+"""
+function add_pseudoribosome!(
+    model::ObjectModel,
+    biomass_rxn_id::String,
+    weight::Float64;
+    pseudoribosome_id = "pseudoribosome",
+)
+    # ensure unidirectional
+    model.reactions[biomass_rxn_id].lower_bound = 0.0
+    model.reactions[biomass_rxn_id].upper_bound = constants.default_reaction_bound
+
+    # add ribosome kinetics
+    model.reactions[biomass_rxn_id].gene_associations = [
+        Isozyme(
+            kcat_forward = weight,
+            kcat_backward = 0.0,
+            gene_product_stoichiometry = Dict(pseudoribosome_id => 1.0),
+        ),
+    ]
+
+    # add ribosome gene
+    model.genes[pseudoribosome_id] = Gene(id = pseudoribosome_id, product_molar_mass = 1.0)
+
+    nothing
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Variant of [`add_pseudoribosome!`](@ref) that does not modify the original
+model, but makes a shallow copy with the modification included.
+"""
+function add_pseudoribosome(
```

Comment:
virtual (too many actual real stuff in biology is called "pseudosomething")

Reply:
so technically virtual is more accurate than pseudo! changing